sale_quotation_products/models/product_product.py

```diff
@@ -77,41 +83,38 @@ def write(self, vals):
             self = self.sudo()
```

Comment:
Multiple improves to this method

1.  > self = self.sudo()

    I have understood that this should not be done. Do not find the source were I read it... As far as I know should be replaced with `super(ModelClass, self.sudo()).write(vals)`

2. > vals.get('qty) and self._context.get('force_product_edit'):

    The dict.get(key) should be always used as dict.get(key, False) to return False in the key is not found and do not raise any error, also because this ones are used in a if condition.

Example

```python
  @api.multi
     def write(self, vals):
         """
         Si en vals solo viene qty y force_product_edit entonces es un dummy
         write y lo hacemos con sudo
         """
         if len(vals) == 1 and vals.get('qty', False) and self._context.get(
                 'force_product_edit', False):
             return super(ProductProduct, self.sudo()).write(vals)
         return super(ProductProduct, self).write(vals)
```